include/albatross/src/utils/linalg_utils.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef ALBATROSS_UTILS_LINALG_UTILS_HPP_
+#define ALBATROSS_UTILS_LINALG_UTILS_HPP_
+
+namespace albatross {
+
+namespace details {
+
+template <typename FeatureType, typename Comparator>
+inline void _print_eigen_directions(const Eigen::MatrixXd &matrix,
+                                    const std::vector<FeatureType> &features,
+                                    std::size_t count, Comparator comp,
+                                    std::ostream *stream) {
+  Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> ev(matrix);
+  const Eigen::VectorXd values = ev.eigenvalues().real();
+  const Eigen::MatrixXd vectors = ev.eigenvectors();
+
+  using ValueAndVector = std::tuple<double, Eigen::VectorXd>;
+  std::vector<ValueAndVector> values_and_vectors;
+
+  for (Eigen::Index i = 0; i < values.size(); ++i) {
+    values_and_vectors.emplace_back(values[i], vectors.col(i));
+  }
+
+  std::sort(values_and_vectors.begin(), values_and_vectors.end(),
+            [&](const ValueAndVector &a, const ValueAndVector &b) -> bool {
+              return comp(std::get<0>(a), std::get<0>(b));
+            });
+
+  (*stream) << "MIN: " << values.minCoeff() << std::endl;
+  (*stream) << "MAX: " << values.maxCoeff() << std::endl;
```

Comment:
Is there a reason to call these methods when values_and_vectors is already sorted?

Reply:
Yeah, the sort order can change from ascending to descending so this makes sure min corresponds to min.